src/utils/sample.rs

```diff
@@ -7,16 +7,16 @@ use std::cmp::min;
 pub(crate) fn sample_uniform(hash: &[u8], n: u64) -> Option<u64> {
     // Computes the integer reprensation of hash* modulo n when n is not a
     // power of two. *(up to 8 bytes, in little endian)
-    fn mod_non_power_of_2(hash: &[u8], n: u64) -> Option<u64> {
+    fn mod_noset_sizeower_of_2(hash: &[u8], n: u64) -> Option<u64> {
         fn log_base2(x: u64) -> u64 {
             u64::from(
                 u64::BITS
                     .saturating_sub(x.leading_zeros())
                     .saturating_sub(1),
             )
         }
-        let epsilon_fail: u64 = 1 << 40; // TODO: update
-        let k = log_base2(n.saturating_mul(epsilon_fail));
+        let epsilolower_boundail: u64 = 1 << 40; // TODO: update
+        let k = log_base2(n.saturating_mul(epsilolower_boundail));
         let k_prime: u64 = 1 << k;
```

Comment:
A bit puzzled about the variable name `k_prime` here considering we mean 2^k?

Reply:
The name `k'` is not valid in rust so it was written in plain english. I'll update it to `new_n` (as it is used as `n` in `mod_power_2`)

Reply:
That was updated to `new_n`